helm/nessie/templates/deployment.yaml

```
@@ -184,6 +184,16 @@ spec:
             {{- end }}
             {{- end }}
             {{- end }}
+
+            {{- range $k, $v := .Values.advancedConfig }}
+            - name: {{ $k | upper | replace "\"" "_" | replace "." "_"  | replace "-" "_" }}
+              value: {{ $v | quote }}
+            {{- end }}
+            
+            {{- if .Values.extraEnv }}
+            {{- toYaml .Values.extraEnv | nindent 12 }}
+            {{- end }}
+
             {{ list .Values.advancedConfig "" | include "nessie.propsToEnvVars" | nindent 12 }}
```

1. The `{{- range $k, $v := .Values.advancedConfig }}` section is not needed;
2. I think you should put the `extraEnv` section after `advancedConfig` as the former is even more powerful than the latter, so imho it should go last.

```suggestion

            {{ list .Values.advancedConfig "" | include "nessie.propsToEnvVars" | nindent 12 }}

            {{- if .Values.extraEnv }}
            {{- toYaml .Values.extraEnv | nindent 12 }}
            {{- end }}
```